nmap.go

```diff
@@ -111,6 +116,42 @@ func (s *Scanner) Run() (*Run, error) {
 	}
 }
 
+// RunAsync runs nmap asynchronously and returns error.
+func (s *Scanner) RunAsync() error {
+	// Enable XML output
+	s.args = append(s.args, "-oX")
+
+	// Get XML output in stdout instead of writing it in a file
+	s.args = append(s.args, "-")
+	s.Cmd = exec.Command(s.binaryPath, s.args...)
+
+	stderr, err := s.Cmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+
+	stdout, err := s.Cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+
+	s.Stdout = *bufio.NewScanner(stdout)
+	s.Stderr = *bufio.NewScanner(stderr)
+
+	if err := s.Cmd.Start(); err != nil {
+		return err
+	}
+
+	go func() {
+		select {
+		case <-s.ctx.Done():
+			s.Cmd.Process.Kill()
```

Comment:
You can write `_ = s.Cmd.Process.Kill()` to explicitly ignore the error :)

Reply:
I'll be updated to the below. Since it is a single select we don't really need to have it written like that. Right?

```go
	go func() {
		<-s.ctx.Done()
		_ = s.Cmd.Process.Kill()
	}()```